chromadb/migrations/migrate_collections.py

```diff
@@ -0,0 +1,103 @@
+# This file contains functions for migrating data from one version of Chroma to another.
+from typing import Optional, Union
+from chromadb.api.client import Client
+from chromadb.api.configuration import (
+    Configuration,
+    ConfigurationParameter,
+)
+from chromadb.segment.impl.vector.hnsw_params import HnswParams
+from pypika import Table
+from chromadb.db.base import ParameterValue, get_sql
+from tqdm import tqdm
+
+
+def migrate_collections(
+    client: Client,
+) -> None:
+    """Migrates the collections in a Local Chroma instance to the latest version of ChromaDB.
+    This is non-destructive and idempotent. It will only update collections that need to be updated.
+    """
+
+    # region Add CollectionConfiguration to Collections with None configuration
+
+    # Monkeypatch the CollectionConfiguration from_json_str to handle None
+    from chromadb.api.configuration import CollectionConfiguration
+
+    original_from_json_str = CollectionConfiguration.from_json_str
+
+    class EmptyConfiguration(Configuration):
+        definitions = {}
+
+    def patched_from_json_str(
```

Comment:
I think we may be able to run this migration as a sql migration? I'm not really a fan at all of this UX, forcing people to run a migration tool for a change like this feels heavy

Reply:
doing this in SQL is scary because it would require direct string manipulation without validation.